osu.Game/Online/SignalRWorkaroundTypes.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using osu.Game.Online.Multiplayer;
+using osu.Game.Online.Multiplayer.MatchTypes.TeamVersus;
+
+namespace osu.Game.Online
+{
+    /// <summary>
+    /// A static class providing the list of types requiring workarounds for serialisation in SignalR.
+    /// </summary>
+    /// <seealso cref="SignalRUnionWorkaroundResolver"/>
+    /// <seealso cref="SignalRDerivedTypeWorkaroundJsonConverter"/>
+    internal static class SignalRWorkaroundTypes
+    {
+        internal static readonly IReadOnlyList<(Type derivedType, Type baseType)> BASE_DERIVED = new[]
+        {
+            (typeof(ChangeTeamRequest), typeof(MatchUserRequest)),
+            (typeof(TeamVersusRoomState), typeof(MatchRoomState)),
+            (typeof(TeamVersusUserState), typeof(MatchUserState)),
+            (typeof(MatchServerEvent), typeof(MatchServerEvent)),
```

Comment:
There doesn't seem to be a reason for having this still existing but didn't want to remove it and potentially break something somehow.

Reply:
That would be because this is not yet used / is planned for future usage, from what I recall. It being here as written is pretty weird, but sure, I guess?

Reply:
I don't think it should be here. As long as there are no derived classes it is not going to be used in any way, and having it in this list is confusing if the same type is listed twice.